topo/topo.go

```diff
@@ -358,17 +402,26 @@ func (m *Manager) Delete(ctx context.Context) error {
 		return fmt.Errorf("topology %q does not exist in cluster", m.proto.Name)
 	}
 
-	// Delete topology pods
+	// Delete topology nodes
 	for _, n := range m.nodes {
 		// Delete Service for node
 		if err := n.Delete(ctx); err != nil {
 			log.Warnf("Error deleting node %q: %v", n.Name(), err)
 		}
-		// Delete Topology for node
-		if err := m.tClient.Topology(m.proto.Name).Delete(ctx, n.Name(), metav1.DeleteOptions{}); err != nil {
-			log.Warnf("Error deleting topology %q: %v", n.Name(), err)
+	}
+	// Delete meshnet nodes
+	nodes, err := m.TopologyResources(ctx)
```

Comment:
i would suggest we should move this delete just into the node it should own all of it's cleanup

Reply:
Wouldn't it make sense for topo manager to delete meshnet resources, since it also creates it ? Moreover, to be able to do this, we'll need to expose `tClient` for `node`s too.

Do let me know what you think. Meanwhile, I tried to make it cleaner to factor out meshnet resource creation/deletion.

https://github.com/google/kne/pull/138/commits/55e3bdcbf3c8b364e7a94de052a968c7a7cee942

Reply:
ack i will need think about this more your point requiring passing the tclient around feels not great